File: php/features.php
<?php
declare(strict_types=1);

// Api2newsEndpoint SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class Api2newsEndpointFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new Api2newsEndpointBaseFeature();
            case "test":
                return new Api2newsEndpointTestFeature();
            default:
                return new Api2newsEndpointBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "test":
                return true;
            default:
                return false;
        }
    }
}

File: py/api2newsendpoint_sdk/features.py
# Api2newsEndpoint SDK feature factory

from api2newsendpoint_sdk.feature.base_feature import Api2newsEndpointBaseFeature
from api2newsendpoint_sdk.feature.test_feature import Api2newsEndpointTestFeature


def _make_feature(name):
    features = {
        "base": lambda: Api2newsEndpointBaseFeature(),
        "test": lambda: Api2newsEndpointTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
